rockcraft/commands/init.py

```diff
@@ -60,28 +129,34 @@ class InitCommand(BaseCommand):
         """
     )
 
-    _INIT_TEMPLATE_YAML = textwrap.dedent(
-        """\
-            name: my-rock-name # the name of your ROCK
-            base: ubuntu:22.04 # the base environment for this ROCK
-            version: '0.1' # just for humans. Semantic versioning is recommended
-            summary: Single-line elevator pitch for your amazing ROCK # 79 char long summary
-            description: |
-                This is my my-rock-name's description. You have a paragraph or two to tell the
-                most important story about it. Keep it under 100 words though,
-                we live in tweetspace and your description wants to look good in the
-                container registries out there.
-            license: GPL-3.0 # your application's SPDX license
-            platforms: # The platforms this ROCK should be built on and run on
-                amd64:
-
-            parts:
-                my-part:
-                    plugin: nil
-            """
-    )
+    def fill_parser(self, parser):
+        """Specify command's specific parameters."""
+        parser.add_argument(
+            "--name", help="The name of the charm; defaults to the directory name"
+        )
+        parser.add_argument(
+            "--profile",
+            choices=list(TEMPLATES),
+            default=DEFAULT_PROFILE,
+            help=f"Use the specified project profile (defaults to '{DEFAULT_PROFILE}')",
+        )
 
     @overrides
     def run(self, parsed_args: "argparse.Namespace") -> None:
         """Run the command."""
-        init(self._INIT_TEMPLATE_YAML)
+        if not parsed_args.name:
+            parsed_args.name = pathlib.Path.cwd().name
+            emit.debug(f"Set project name to '{parsed_args.name}'")
+
+        if not re.match(r"[a-z][a-z0-9-]*[a-z0-9]$", parsed_args.name):
```

Comment:
why is this regex different from

https://github.com/canonical/rockcraft/blob/318ecdf85872e9d0769d2b62e70fea4acc77505d/rockcraft/project.py#L159

? 
In any case, I don't think it should return an error. If the parameter is optional, then the command should always work whenever that option is not provided. 

I'd instead suggest issuing a warning and populating the name with the current default `my-rock-name # the name of your ROCK`

Reply:
Sorry the name regex is charmcraft's, I have updated that. I've also update the name error handling: if a user explicitly provides an invalid name, it will trigger an error. However, if the name is inferred implicitly, it will default to `my-rock-name` without causing an error.

Reply:
Thanks! You might want to add tests for those scenarios.

Reply:
Tests added, thanks!